addons/account/models/account_move_line.py

```diff
@@ -1547,13 +1550,22 @@ def changed(fname):
                     line.balance = amount_currency
 
         after = existing()
+        changed_currency_rate = any(changed_existing_val('currency_rate') for line in after)
+        term_line = False
+        aggregated_balance = 0
         for line in after:
+            if line.display_type == 'payment_term' and changed_currency_rate and len(after) > 1:
```

Comment:
Why `and len(after) > 1`?

Reply:
The `sync_invoice` method is called on individual `account.move.lines` during a `create` call and on the full recordset during a write call. When the `payment_term` line is being created it will erroneously be written to have a balance of 0 without this check.